### ISSUE TITLE
openssl is still required

### ISSUE THREAD

**Issue by dario23** [smcameron/space-nerds-in-space#111]
commit 6e2602a5 removes openssl dependencies, claiming that it isn't required anymore. i think it is in fact still required:

```
> git grep '^#include <openssl'
snis_hash.c:#include <openssl/md5.h>
snis_hash.c:#include <openssl/sha.h>
```

and building a debian package (at least when using sbuild or pbuilder, where openssl incidentally being installed in the host system doesn't save the build like when using debuild or dpkg-buildpackage) fails when compiling the snis_hash.c file.

i think the solution to this would be to re-add the openssl dependencies in `debian/control` and the relevant parts in `Makefile`, but it the medium to long term either the used parts (md5 and sha*) should be replaced or another solution to the licensing issues with openssl be found (what exactly is the problem here? and how do other GPL projects solve that?)

**Comment by smcameron** [comment]
Does commit f72f90b342dbb42a5b410b3e0f06b7994539e43e fix this?

**Comment by smcameron** [comment]
Hmm, after looking into this more, and at the commit message of 6e2602a which links to:
https://people.gnome.org/~markmc/openssl-and-the-gpl.html and actually looking at that, it turns out that openssl and the gpl have some sort of incompatibility problem.  So this is going to need some more work.

**Comment by dario23** [comment]
@smcameron yes to both ;) builds again here, but the openssl/gpl-exception is still neccessary, or replacing the relevant parts to avoid linking libssl in the longer term.

**Comment by smcameron** [comment]
Ok, I think this should be fixed by fb4a668f190b65bf787908fd930c826f1064479f and 102e76a4bd8157fff4bbde7d10da6f1584733a50

Used crypt_r instead.
